apps/mull-ui/src/app/pages/profile/edit-profile/edit-profile.tsx

```diff
@@ -1,57 +1,88 @@
+import { useToast } from '../../../hooks/useToast';
 import './edit-profile.scss';
 
-const EditProfile = () => {
-  // TODO: Add initial profile image if user has one
-  // TODO: If user does not have an image, set to a default image
-  const [imageURLFile, setImageURLFile] = useState<string>(
-    'https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg'
-  );
+export interface EditProfilePageProps {
+  history: History;
+}
 
-  // Uploaded image file blob
+const EditProfile = ({ history }: EditProfilePageProps) => {
+  const [imageURLFile, setImageURLFile] = useState<string>('');
   const [file, setFile] = useState<File>(null);
-
-  /**
-   * Handles image file uploads
-   * @param {ChangeEvent<HTMLInputElement>} event
-   */
-  const handleFileUpload = (event: ChangeEvent<HTMLInputElement>) => {
-    setImageURLFile(URL.createObjectURL(event.target.files[0]));
-    setFile(event.target.files[0]);
-    formik.setFieldValue('imageFile', event.target.files[0]);
-  };
+  const { data: userData, loading: userLoading } = useUserQuery();
+  const [updateUser] = useUpdateUserMutation();
+  const { updateToast, notifyToast } = useToast();
 
   const formik = useFormik<IProfileEditForm>({
     initialValues: {
       displayName: '',
       description: '',
-      imageFile: null,
+      imageFile: '',
     },
-
     validationSchema: Yup.object({
       displayName: Yup.string().required('Display name is required'),
-      description: Yup.string().required('Description is required'),
     }),
-    //TODO: implement form logic
+
     onSubmit: async () => {
       const errors = await formik.validateForm();
       if (isEmpty(errors)) {
-        //TODO: update profile in database
-        console.log('Form successfully submitted');
+        notifyToast('Submitting User Profile...');
+        try {
+          await updateUser({
+            variables: {
+              newAvatar: file,
+              userInput: {
+                id: userData.user.id,
+                name: formik.values.displayName,
+                description: formik.values.description,
+              },
+            },
+          });
+          updateToast(toast.TYPE.SUCCESS, 'Profile Updated');
+          history.push(ROUTES.PROFILE.DISPLAY);
+        } catch (err) {
+          updateToast(toast.TYPE.ERROR, 'Error: Failed To Update Profile');
+          console.error(err);
+        }
       } else {
         formik.setTouched(setNestedObjectValues<FormikTouched<FormikValues>>(errors, true));
       }
     },
   });
 
-  return (
+  useEffect(() => {
+    if (!userLoading) {
+      setImageURLFile(avatarUrl(userData.user as User));
+      formik.setFieldValue('displayName', userData.user.name);
+      formik.setFieldValue('description', userData.user.description);
+    }
+    // Ignored since the suggested fix broke the react hooks lifecycle
+    /* eslint-disable-next-line react-hooks/exhaustive-deps */
+  }, [userData]);
```

Comment:
We could silence the linter by doing this. Probably better as well, it's complaining that we didn't add all the necessary dependencies to the array
```suggestion
  }, [userData, userLoading, formik]);
```

Reply:
I actually did add that in at some point, however (I think due to Formik), the page was rendering on an endless loop, which is why I ignored it

Reply:
truee, alright sounds good